scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -494,7 +494,7 @@ protected NodeSet getRMNodes(Map<JobId, JobDescriptor> jobMap, int neededResourc
                 }
             } else {
                 // otherwise use the default job variables
-                variables.putAll(job.getVariables());
+                variables.putAll(task.getVariablesOverriden(job));
```

Comment:
In the createBindingsForSelectionScripts method, the case where task variables override propagated variables should be handled. I don't think the getVariablesOverriden method will be enough for this scenario.